credentials/tls_ext_test.go

```diff
@@ -236,3 +238,99 @@ func (s) TestTLS_CipherSuitesOverridable(t *testing.T) {
 		t.Fatalf("EmptyCall err = %v; want <nil>", err)
 	}
 }
+
+// TestTLS_DisabledALPN tests the behaviour of a gRPC client when connecting to
+// a server that doesn't support ALPN.
+func (s) TestTLS_DisabledALPN(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	initialVal := envconfig.EnforceALPNEnabled
+	defer func() {
+		envconfig.EnforceALPNEnabled = initialVal
+	}()
+
+	// Start a non gRPC TLS server.
+	config := &tls.Config{
+		Certificates: []tls.Certificate{serverCert},
+		NextProtos:   []string{}, // Empty list indicates ALPN is disabled.
+	}
+	listner, err := tls.Listen("tcp", ":0", config)
```

Comment:
We have had some issues in the past with starting a listener with address in `:0` (especially on forge). One option might be to use `localhost:0` here, or to use https://github.com/grpc/grpc-go/blob/5d24ee2bd170a0c2c636b7d1ebc7ebdd3ecf81ac/internal/testutils/local_listener.go#L24 and pass it to https://pkg.go.dev/crypto/tls#NewListener instead.

Reply:
Also, we have had issues in the past where common stuff like this is started once for the whole test instead of once per entry in the test table below (making the sub-tests not completely hermetic). I would recommend moving this to be inside `t.Run`.

Reply:
Moved inside the closure.